source/common/conn_pool/conn_pool_base.cc

```diff
@@ -473,6 +473,14 @@ void ConnPoolImplBase::onConnectionEvent(ActiveClient& client, absl::string_view
     ASSERT(client.state() == ActiveClient::State::CONNECTING);
     transitionActiveClientState(client, ActiveClient::State::READY);
 
+    // Now that the active client is ready, set up a timer for max connection duration.
+    const auto max_connection_duration = client.parent_.host()->cluster().maxConnectionDuration();
```

Comment:
can you use an explicit type here?

Reply:
Added the full type (an optional) and updated the conditional to check has_value()